core-common/src/main/java/org/apache/kylin/common/notify/MailService.java

```diff
@@ -78,12 +73,17 @@ public boolean sendMail(List<String> receivers, String subject, String content)
      * @return true or false indicating whether the email was delivered successfully
      * @throws IOException
      */
-    public boolean sendMail(List<String> receivers, String subject, String content, boolean isHtmlMsg) {
+    private boolean sendMail(List<String> receivers, String subject, String content, boolean isHtmlMsg) {
 
         if (!enabled) {
             logger.info("Email service is disabled; this mail will not be delivered: " + subject);
             logger.info("To enable mail service, set 'kylin.job.notification-enabled=true' in kylin.properties");
             return false;
+        } else {
+            if (host.isEmpty()) {
```

Comment:
Can we extract this if block?  There is no need to have the `else` at line No.82.